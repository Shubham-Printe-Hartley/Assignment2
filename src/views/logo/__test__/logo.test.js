import React from 'react';
import ReactDOM from 'react-dom';
import Logo from '../logo';
import logoImg from '../../../assets/logo2.png'

import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

// run this after each test
afterEach(cleanup);

// checks if logo gets rendered without crashing
it("renders logo without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Logo></Logo>, div);
});

// checks if logo has the required properties
it("renders logo correctly", () => {
  const {getByTestId} = render(<Logo img={logoImg}></Logo>);
  expect(getByTestId('logo')).toHaveAttribute('alt', 'logo');
  expect(getByTestId('logo')).toHaveAttribute('src', 'logo2.png');
});

// snapshot testings - takes a snapshot of the component and records it in the first run
// compares this snapshot in the subsequent tests
it("matches snapshot 1", () => {
  const logoTree = renderer.create(<Logo img={logoImg} ></Logo>).toJSON();
  expect(logoTree).toMatchSnapshot();
});







