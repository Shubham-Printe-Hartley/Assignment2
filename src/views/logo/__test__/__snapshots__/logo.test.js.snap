// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="logo-wrapper"
>
  <img
    alt="logo"
    data-testid="logo"
    src="logo2.png"
  />
</div>
`;
